node/src/filecheckpointer.ts

```diff
@@ -73,12 +79,13 @@ export class FileCheckPointer implements Checkpointer {
 
     #setState(state: CheckpointerState): void {
         this.#blockNumber = state.blockNumber ? BigInt(state.blockNumber) : undefined;
-        this.#transactionIDs = new Set(state.transactionIDs);
+        this.#transactionID = state.transactionID;
     }
+
     #getState(): CheckpointerState {
         return {
             blockNumber: (this.#blockNumber !== undefined) ? this.#blockNumber.toString() : undefined,
```

Comment:
You might consider taking advantage of the `?.` operator
```suggestion
            blockNumber: this.#blockNumber?.toString(),
```